### DISCUSSION TITLE
how to model mixed-integer linear program in casadi

### DISCUSSION THREAD

**Discussion by HQBao1994** [casadi/casadi#3231] [Q&A]
Hi,
Everyone, how can i model mixed-integer linear program in casadi.
Is there any example?
for example, how to code the binary variable (0 or 1) in casadi

**Comment by jgillis** [comment]
https://github.com/casadi/casadi/wiki/FAQ:-how-to-work-with-discrete-mixed-integer-problems%3F may provide an answer. If not, let us know..

**Reply by HQBao1994** [reply]
Hi, 
Thanks a lot!
I have one question, for example, x is a binvar(0,1), so it means that I need set  x's discrete = [true] and  0=<x<=1?

**Reply by jgillis** [reply]
That's right; feel free to edit the wiki page.

**Reply by saruchrathore** [reply]
Thanks for the wiki.
I am also trying to solve a MILP. The example works in MATLAB but not in python. I am getting error 'Plugin bonmin is not found'. How do I get the bonmin solver? It seems to be missing '...bonmin.dll'.
